diff --git a/packages/proot/build.sh b/packages/proot/build.sh
index 22c42ed994ad4..57c53615725b9 100644
--- a/packages/proot/build.sh
+++ b/packages/proot/build.sh
@@ -8,7 +8,7 @@ TERMUX_PKG_VERSION=5.1.107
 TERMUX_PKG_REVISION=54
 TERMUX_PKG_SRCURL=https://github.com/termux/proot/archive/${_COMMIT}.zip
 TERMUX_PKG_SHA256=2c2a24386587dc852582774169c8f981f043b63630339db0e6046de8ddb876b7
-TERMUX_PKG_DEPENDS="libtalloc"
+TERMUX_PKG_DEPENDS="libtalloc-static, libtalloc"
 TERMUX_PKG_BUILD_IN_SRC=true
 TERMUX_PKG_EXTRA_MAKE_ARGS="-C src"
 
@@ -16,6 +16,8 @@ TERMUX_PKG_EXTRA_MAKE_ARGS="-C src"
 export PROOT_UNBUNDLE_LOADER=$TERMUX_PREFIX/libexec/proot
 
 termux_step_pre_configure() {
+	LDFLAGS+=" -static"
+	LDFLAGS+=" -ffunction-sections -fdata-sections -Wl,--gc-sections"
 	CPPFLAGS+=" -DARG_MAX=131072"
 }
 
